static/js/actions/email.js

```diff
@@ -31,6 +31,23 @@ export const sendEmailSuccess = createAction(SEND_EMAIL_SUCCESS);
 export const SEND_EMAIL_FAILURE = 'SEND_EMAIL_FAILURE';
 export const sendEmailFailure = createAction(SEND_EMAIL_FAILURE);
 
+export function sendEmail(
+  emailType: string,
+  sendFunction: () => Promise<EmailSendResponse>,
+  sendFunctionParams: Array<*>
+): Dispatcher<EmailSendResponse> {
+  return (dispatch: Dispatch) => {
+    dispatch(initiateSendEmail(emailType));
+    return sendFunction.apply(null, sendFunctionParams).
+      then(response => {
+        dispatch(sendEmailSuccess(emailType));
+        return Promise.resolve(response);
+      }).catch(error => {
+        dispatch(sendEmailFailure({type: emailType, error: error}));
```

Comment:
Should this return `Promise.reject(error)`? And is there any UI to notify the user in case the email fails to send? A toast message would suffice.